src/main/java/com/fluxninja/aperture/armeria/DecoratingHTTPClient.java

```diff
@@ -0,0 +1,69 @@
+package com.fluxninja.aperture.armeria;
+
+import com.fluxninja.aperture.sdk.ApertureSDK;
+import com.fluxninja.aperture.sdk.ApertureSDKException;
+import com.fluxninja.aperture.sdk.Flow;
+import com.fluxninja.aperture.sdk.FlowStatus;
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.client.SimpleDecoratingHttpClient;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DecoratingHTTPClient extends SimpleDecoratingHttpClient {
+    private ApertureSDK apertureSDK;
+    private String featureName;
+
+    public DecoratingHTTPClient(HttpClient delegate, ApertureSDK apertureSDK, String featureName) {
```

Comment:
No need to take featureName from user. Can leave featureName empty for now. Will eventually move to Envoy Authz protocol